### ISSUE TITLE
[@types/joi] cannot use with joi@15 cause it imports "@types/joi": "*"

### ISSUE THREAD

**Issue by kirillgroshkov** [DefinitelyTyped/DefinitelyTyped#39169]
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/joi` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @Bartvds @laurence-myers @cglantschnig @DavidBR-SW

Issue described there: https://github.com/hapijs/joi/issues/2184

Basically, `package.json` has `"dependencies": {"@types/hapi__joi": "*"}` which breaks the usage of `@hapi/joi@15`, cause it installs `16` with `*`.

Started breaking after `@types/hapi_joi@16` was published.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
